libpod/runtime_ctr.go

```diff
@@ -54,6 +54,14 @@ func (r *Runtime) RestoreContainer(ctx context.Context, rSpec *spec.Spec, config
 	}
 	// For an imported checkpoint no one has ever set the StartedTime. Set it now.
 	ctr.state.StartedTime = time.Now()
+
+	// If the path to ConmonPidFile starts with the default value (RunDir), then
+	// the user has not specified '--conmon-pidfile' during run or create (probably).
+	// In that case reset ConmonPidFile to be set to the default value later.
+	if strings.HasPrefix(ctr.config.ConmonPidFile, ctr.state.RunDir) {
+		ctr.config.ConmonPidFile = ""
```

Comment:
I think we do expect this to be set for all containers now.

Reply:
Actually, why is this a problem? ctr.state.RunDir should be unique for all containers?

Reply:
My goal was to make sure if the user selected a specific conmon.pid during start with `--conmon-pidfile /tmp/conmon1.pid`, that the restored container uses the same file.

Reply:
Ahhh. Are you resetting it on the expectation that it will be populated again during container creation by a fresh value?

Reply:
Exactly. If it is an empty string it will be set to the default value for this newly restored container. But I just saw my testing was not correct. I need to do it at another place in the code, because at this point in time RunDir is not yet set.

Thinking about it more, it is problematic. If the user selects a specific conmon pidfile it will be overwritten if the container is restored multiple times. It is basically the same problem if a user starts two containers and points them to the same file for conmon.pid.

If `--conmon-pidfile` is used the user must know that the file will be overwritten if using the same file multiple times.